multisig/src/lib.rs

```diff
@@ -192,6 +275,36 @@ impl MultiSigContract {
             .map(|key| Base58PublicKey::try_from(key).expect("Failed to covert key to base58"))
             .collect()
     }
+
+    pub fn on_transaction_completed(&mut self, request_id: RequestId, transaction_index: usize) -> bool {
+        assert_eq!(env::predecessor_account_id(), env::current_account_id(), "Callback can only be called from the contract");
+        let success = is_promise_success();
+        assert!(success, format!("Transaction {} in request {} has failed", transaction_index, request_id));
```

Comment:
that's fine, isn't it? I'm not sure this should proceed if there is a failure.
It's actually unclear how exactly chaining should be done in cases of possible failures, but i think normal expectation that if anything failed - it should not continue.
If the recipient returned promise, instead of handling it with callback - they wanted to return this failure to the caller IMO.

Reply:
I think we should handle multiple transactions on the client side instead. It doesn't make sense to do this in a single call.